app/src/main/java/cash/z/ecc/android/ui/profile/ProfileViewModel.kt

```
@@ -94,33 +111,57 @@ class ProfileViewModel @Inject constructor() : ViewModel() {
     }
 
     suspend fun fullRescan() {
-        rewindTo(synchronizer.latestBirthdayHeight)
+        synchronizer.latestBirthdayHeight?.let {
+            rewindTo(it)
+        }
```

Do we want to throw an exception or something here in case `latestBirthdayHeight` is null?

---

I think previously it could try to rewind to -1, which I suspect has undefined behavior.

I agree that fail-fast would be an enhancement.